bot/src/stake_pool.rs

```diff
@@ -1075,6 +1075,7 @@ mod test {
             &authorized_staker,
             &authorized_staker.pubkey(),
             num_validators,
+            &withdraw_authority,
         )
         .unwrap();
```

Comment:
This test fails here with the following error:

```
thread 'stake_pool::test::this_test_is_too_big_and_slow' panicked at 'called `Result::unwrap()` on an `Err` value: ClientError { request: Some(SendTransaction), kind: RpcError(RpcResponseError { code: -32002, message: "Transaction simulation failed: Error processing Instruction 2: incorrect program id for instruction", data: SendTransactionPreflightFailure(RpcSimulateTransactionResult { err: Some(InstructionError(2, IncorrectProgramId)), logs: Some(["Program 11111111111111111111111111111111 invoke [1]", "Program 11111111111111111111111111111111 success", "Program 11111111111111111111111111111111 invoke [1]", "Program 11111111111111111111111111111111 success", "Program SPoo1Ku8WFXoNDMHPsrGSTSG1Y47rzgn41SLUNakuHy invoke [1]", "Program log: Instruction: Initialize stake pool", "Program log: Expected account to be owned by program SPoo1Ku8WFXoNDMHPsrGSTSG1Y47rzgn41SLUNakuHy, received 11111111111111111111111111111111", "Program log: Error: IncorrectProgramId", "Program SPoo1Ku8WFXoNDMHPsrGSTSG1Y47rzgn41SLUNakuHy consumed 21512 of 600000 compute units", "Program SPoo1Ku8WFXoNDMHPsrGSTSG1Y47rzgn41SLUNakuHy failed: incorrect program id for instruction"]), accounts: None, units_consumed: Some(0) }) }) }', bot/src/stake_pool.rs:1080:10
```

The instruction that is failing is `spl_stake_pool::instruction::initialize`; presumably that instruction is getting constructed incorrectly.

Reply:
You can see the full test output here: https://github.com/solana-foundation/stake-o-matic/actions/runs/3905616039/jobs/6672832690